src/backend/catalog/dependency.c

```diff
@@ -950,9 +952,85 @@ findDependentObjects(const ObjectAddress *object,
 		dependentObjects[numDependentObjects].subflags = subflags;
 		numDependentObjects++;
 	}
-
+	is_enr = scan->enr;
 	systable_endscan(scan);
 
+	/* Indices for temp tables will leave metadata in pg_catalog, so we should clean them up here. */
+	if (is_enr)
```

Comment:
I am wondering why is this needed if code is supposed to add temp table index to ENR? If index is in ENR then we wouldnt need this, no?

Reply:
This is needed because the below code does add the pg_index entries to ENR, but doesn't prevent them from being created in catalogs. It's those entries in pg_class, pg_depend etc that will not be cleaned up in findDependentObjects.